Disparity to depth

GerryChen117 opened the pull request.

GerryChen117 committed:
trying something haha

GerryChen117 committed:
Revert "trying something haha"

This reverts commit 0474474a64e9bc840b25a0b841b92b0d3a7f767b.

GerryChen117 committed:
Separate file for obstacle locating

- Locates obstacles in world frame and publishes to a topic

GerryChen117 committed:
Added Bridge

cheongb002 commented on src/constants.py:
```diff
@@ -24,6 +24,7 @@
 VICON_POSE_TOPIC = "/vicon/ROB498_Drone/ROB498_Drone"
 WAYPOINTS_TOPIC = "/{}/comm/waypoints".format(NAME)
 MAVROS_SETPOINT_TOPIC = "/mavros/setpoint_position/local"
+DISPARITY_MAP_TOPIC = "/camera/stereo/image_raw"
```
Verify that this is actually the topic name

cheongb002 commented on src/obstacle_detection_node.py:
```diff
@@ -0,0 +1,264 @@
+
+def update_drone_pose(msg):
+    global DRONE_POSE
+    DRONE_POSE = msg
```
Switch to using the tf tree to determine the drone's position

cheongb002 commented on src/obstacle_detection_node.py:
```diff
@@ -0,0 +1,264 @@
+    Q_MATRIX = Q
+
+
+def process_point_cloud(disparity, Q_MATRIX):
```
we can actually get a point cloud by directly subscribing to /camera/pointcloud2

cheongb002 commented on src/obstacle_detection_node.py:
```diff
@@ -0,0 +1,264 @@
+    point_cloud = point_cloud[mask]
+
+    # Step 3: Find all points at y = 0
+    mask = point_cloud[:,:,1] == 0
```
if we run this before the distance mask (step 2) we can save computation

cheongb002 commented on src/obstacle_detection_node.py:
```diff
@@ -0,0 +1,264 @@
+    point_cloud = point_cloud[mask]
+
+    # Step 4: Find the closest point to the drone
+    min_distance = np.min(point_cloud[:,:,2])
```
shouldn't min dist be min euclidean distance to the drone?

cheongb002 commented on src/obstacle_detection_node.py:
```diff
@@ -0,0 +1,264 @@
+    located_obstacle_pose_stamp = locate_obstacles(closest_obstacles)
+
+    # Step 6: Transform the located obstacle to world frame
+    world_obstacle_pose_stamp = transform_map_to_world(located_obstacle_pose_stamp)
```
We can actually do all of the above computation in world space to begin with (since we should have the drone position in world space and we can transform the PC to world space using the TF tree)

cheongb002 commented on src/obstacle_detection_node.py:
```diff
@@ -0,0 +1,264 @@
+
+
+
+#===============================================================================
```
Remove the old code

cheongb002 commented on src/obstacle_detection_node.py:
```diff
@@ -0,0 +1,264 @@
+def camera_info_cb1(msg):
+    global CAM_INFO1
+    CAM_INFO1 = copy.deepcopy(msg)
+
```
we should add a check to see if cam_info is none. If it is not, we don't need to run this deep copy again since it should be constant

cheongb002 commented on src/obstacle_detection_node.py:
```diff
@@ -0,0 +1,264 @@
+    return pose_stamped
+
+
+# TODO: Functions for transforming points and detecting obstacles
```
remove the todo if finished

cheongb002 commented on src/obstacle_detection_node.py:
```diff
@@ -0,0 +1,264 @@
+# TODO: Functions for transforming points and detecting obstacles
+def transform_map_to_world(pose_array):
+    # Transform points in PoseArray from map to world frame
+    transformed_pose_array = PoseArray()
```
if we only have a single point detected from the PC, we don't need a pose array

GerryChen117 committed:
Merge branch 'main' into disparity_to_depth

GerryChen117 committed:
Update using tf_tree

- Obstacles are found in point cloud frame, then directly converted to world frame using tf tree

GerryChen117 committed:
Revised 3:48pm April 13th

Commit:
fix tf buffer issue

Commit:
add frame dropping to undistortion

GerryChen117 committed:
Synchronization fixes

GerryChen117 committed:
Final updates

Commit:
tested changes to wf,
